### PR TITLE
refactor(forge): Clean `forge`

### PR DESCRIPTION
**Pull request by kcze**
### Background

Should be merged after:
- https://github.com/Significant-Gravitas/AutoGPT/pull/7106
- ~~https://github.com/Significant-Gravitas/AutoGPT/pull/7151~~

Follow-up after [Component-based Agents](https://github.com/Significant-Gravitas/AutoGPT/pull/7054)

### Changes 🏗️

**TL;DR:** remove unused `forge` code and improve structure of `forge`.

* Put all Agent Protocol stuff together in `forge.agent_protocol`
  * ... including `forge.agent_protocol.database` (was `forge.db`)
* Remove duplicate/unused parts from `forge`
  * `forge.actions`, containing old commands; replaced by `forge.components` from `autogpt`
  * `forge/agent.py` (the old one, `ForgeAgent`)
  * `forge/app.py`, which was used to serve and run the `ForgeAgent`
  * `forge/db.py` (`ForgeDatabase`), which was used for `ForgeAgent`
  * `forge/llm.py`, which has been replaced by new `forge.llm` module which was ported from `autogpt.core.resource.model_providers`
  * `forge.memory`, which is not in use and not being maintained
  * `forge.sdk`, much of which was moved into other modules and the rest is deprecated
  * `AccessDeniedError`: unused
  * `forge_log.py`: replaced with `logging`
  * `validate_yaml_file`: not needed
  * `ai_settings_file` and associated loading logic and env var `AI_SETTINGS_FILE`: unused
  * `prompt_settings_file` and associated loading logic and env var `PROMPT_SETTINGS_FILE`: default directives are now provided by the `SystemComponent`
    * `request_user_double_check`, which was only used in `AIDirectives.load`
  * `TypingConsoleHandler`: not used

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
enhancement, bug fix, tests

___

### **Description**
- Refactored the `forge` module to improve structure and remove unused code.
- Consolidated all Agent Protocol related code into `forge.agent_protocol`.
- Removed duplicate and deprecated modules such as `forge.actions`, `forge/agent.py`, `forge/app.py`, `forge/db.py`, `forge/llm.py`, `forge/memory`, `forge/sdk`, `AccessDeniedError`, `forge_log.py`, and `validate_yaml_file`.
- Updated import paths across various modules to reflect the new structure.
- Replaced custom logging with standard `logging` module.
- Removed YAML validation logic from multiple files.
- Updated and removed related test cases.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>agent_protocol_server.py</strong><dd><code>Refactor imports and remove unused validation in </code><br><code><code>agent_protocol_server</code>.</code></dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/agent_protocol_server.py

<li>Updated imports to reflect new module structure.<br> <li> Removed unused import <code>validate_yaml_file</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-8687cb793322c5a9217f5182d796e3befbcb6c37387f5614f719c514874d5a3b">+9/-9</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>configurator.py</strong><dd><code>Remove YAML validation and update imports in `configurator`.</code></dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/configurator.py

- Removed YAML file validation logic.
- Updated import statements.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-df049ca064a119e4a3bbc1c4f91f7b3811f4568b0a92225054032034dbfed47d">+2/-21</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Update import paths and remove redundant import in `main`.</code></dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/main.py

- Updated import paths for `AgentDB`.
- Removed redundant import.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-3ca14b0308a80bf8ccf59bfce0ab9fd9ec4bc8223b14493a24f57b9150e704de">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Remove YAML validation function in `utils`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/utils.py

- Removed `validate_yaml_file` import and related function.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-dbeaac8c7ae77c7805c9962ec2716a66a412418509e992123c7a50b497792b1c">+1/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>__main__.py</strong><dd><code>Replace custom logger with standard logging in `__main__.py`.</code></dd></summary>
<hr>
      
autogpts/forge/forge/__main__.py

- Replaced custom logger with standard `logging`.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-2f2c41a61ec3c390d4712b2ae9a3b5599e7255800f97bd4e0a5e6aa382485ba0">+3/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>api_router.py</strong><dd><code>Update imports and replace custom logger in `api_router`.</code></dd></summary>
<hr>
      
autogpts/forge/forge/agent_protocol/api_router.py

<li>Updated import paths.<br> <li> Replaced custom logger with standard <code>logging</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-491b0b16813a9e77e1bf4fa7815d965fac00d12921bb97a21964defbd6794bbb">+25/-25</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>db.py</strong><dd><code>Update imports and replace custom logger in `db`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/agent_protocol/database/db.py

<li>Updated import paths.<br> <li> Replaced custom logger with standard <code>logging</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-7bb8dc26229ec399cc8e9416750da56a1f6ed624a1c22e1dbbfcde85800fec06">+51/-47</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>task.py</strong><dd><code>Refactor models and rename `Status` to `StepStatus`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/agent_protocol/models/task.py

<li>Moved <code>Artifact</code> and <code>Pagination</code> models to separate files.<br> <li> Renamed <code>Status</code> to <code>StepStatus</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-6acf5f107956d0b962c3f4692b789cbef800f10b1052393504e357513c2d2f5e">+4/-60</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>middlewares.py</strong><dd><code>Simplify import statements in `middlewares`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/agent_protocol/middlewares.py

- Simplified import statements.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-732cd236c02653ee0565f83b3f10c4be46a7285ac26c0d13d93442b12e1350c6">+2/-7</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>parameter.py</strong><dd><code>Replace `dataclasses` with `pydantic.BaseModel` in `parameter`.</code></dd></summary>
<hr>
      
autogpts/forge/forge/command/parameter.py

- Replaced `dataclasses` with `pydantic.BaseModel`.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-1e5a36de9495eff8d3c860c993892053f47c50c8beb2b590ac7975d8105ecf70">+2/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>code_executor.py</strong><dd><code>Add `CodeExecutionError` class in `code_executor`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/components/code_executor/code_executor.py

- Added `CodeExecutionError` class.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-da228cb73fb98d6a07e27299efde7aa9249a2a31f4136c049811e7a6795a80ad">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ai_directives.py</strong><dd><code>Remove YAML validation logic in `ai_directives`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/config/ai_directives.py

- Removed YAML validation logic.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-4c2d7e2a559100fa70087295dc318cd578b98f4675d24ed8cab5f411823cc35c">+0/-10</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ai_profile.py</strong><dd><code>Remove YAML file handling methods in `ai_profile`.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/forge/forge/config/ai_profile.py

<li>Removed methods for loading and saving AI profiles from YAML files.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-dd9c464bf444726b7147cced6d1fc32466e8a01a908c33b1266f8adf6b5e31eb">+0/-50</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests
</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test_ai_profile.py</strong><dd><code>Remove `AIProfile` related test cases.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/tests/unit/test_ai_profile.py

- Removed test cases related to `AIProfile` class.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-b5d3129efa3c68fd8d2745c5cf9d6f251b7c46adcd47dbb4ceece9b02cff598e">+0/-71</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>test_utils.py</strong><dd><code>Remove `validate_yaml_file` test cases.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/tests/unit/test_utils.py

- Removed test cases for `validate_yaml_file`.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7117/files#diff-522c92f2b0d1a4445df47605c6940f023d0eefddbe6886eb03478a1daf950f8c">+0/-36</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: autogpts/forge/forge/__main__.py [hunk 1]
```diff
@@ -36,9 +35,8 @@
 if __name__ == "__main__":
     print(logo)
     port = os.getenv("PORT", 8000)
-    LOG.info(f"Agent server starting on http://localhost:{port}")
+    logger.info(f"Agent server starting on http://localhost:{port}")
```
Comment (Pwuts) [inline autogpts/forge/forge/__main__.py:41]: this won't do anything unless the logger is configured, e.g. using `configure_logging`
```suggestion
    configure_logging()
    logger.info(f"Agent server starting on http://localhost:{port}")
```

File: autogpts/forge/forge/components/web/selenium.py [hunk 2]
```diff
@@ -49,6 +49,10 @@ class BrowsingError(CommandExecutionError):
     """An error occurred while trying to browse the page"""
 
 
+class TooMuchOutputError(CommandExecutionError):
+    """The operation generated more output than what the Agent can process"""
+
+
```
Comment (Pwuts) [inline autogpts/forge/forge/components/web/selenium.py]: IMO this is a general error and shouldn't be tied to this specific component.

File: autogpts/forge/forge/config/ai_directives.py [hunk 3]
```diff
@@ -22,26 +20,7 @@ class AIDirectives(BaseModel):
...
     best_practices: list[str] = Field(default_factory=list)
 
-    @staticmethod
-    def from_file(prompt_settings_file: Path) -> "AIDirectives":
-        from forge.logging.helpers import request_user_double_check
-
-        (validated, message) = validate_yaml_file(prompt_settings_file)
-        if not validated:
-            logger.error(message, extra={"title": "FAILED FILE VALIDATION"})
-            request_user_double_check()
```
Comment (Pwuts) [inline autogpts/forge/forge/config/ai_directives.py:32]: With this usage gone, iirc we can remove `request_user_double_check` from `forge`. Maybe it's still used in `autogpt` but then it should be defined there.

File: autogpts/forge/forge/components/web/selenium.py (hunk 2 above)
Comment (kcze) [reply autogpts/forge/forge/components/web/selenium.py]: Moved back to `exceptions.py`

File: autogpts/forge/forge/config/ai_directives.py (hunk 3 above)
Comment (kcze) [reply autogpts/forge/forge/config/ai_directives.py:32]: When I remove `request_user_double_check`, then `user_friendly_output` is also not used, so removing 🧹
Also moved `print_attribute` and `speak` to `logging/utils.py`, so I could remove `logging/helpers.py`.

Comment (Pwuts) [reply autogpts/forge/forge/config/ai_directives.py:32]: if `user_friendly_output` is not used, then the `typing_console_handler` and `user_friendly_output_logger` also are not used afaik :)

**Review by Pwuts — APPROVED** [review]

**Pull request merged**